geoProcess.py

```diff
@@ -0,0 +1,275 @@
+                # determine whether or not the input type is geometrycollection. If so, invoke an unwrap function
+                if data['features'][index]['geometry']['type'] == 'GeometryCollection':
+                    if len(data['features'][index]['geometry']['geometries']) != 0:
+                        # iterates through all elements in "geometries" and find the bounding box
+                        tmp_bounding_box, tmp_geometry_collec = unwrap_func(data['features'][index]['geometry']['geometries'],
+                                                                            data['features'][index]['properties']['feature_properties'],
+                                                                            int_array[index], f)
+                        # ==============================
+                        #for element in tmp_geometry_collec:
+                        #    output_data.append(element)
+                        output_data = output_data + tmp_geometry_collec
+                    # ==============================
+                else:
+                    # discard a feature without its feature property
+                    if len(data['features'][index]['properties']['feature_properties']) != 0:
+                        tmp_bounding_box= min_max_calculation(data['features'][index]['geometry']['type'],
+                                                              data['features'][index]['geometry']['coordinates'])
+                        output_data.append([data['features'][index]['geometry']['type'], data['features'][index]['geometry']['coordinates'],
+                                            int_array[index], data['features'][index]['properties']['feature_properties'][0]['osmid'],
+                                            data['features'][index]['properties']['feature_properties'][0]['ItemId'], f])
+                # update the minimum and maximum values
+                bounding_box[0] = update_function(bounding_box[0], tmp_bounding_box[0], 0)
+                bounding_box[1] = update_function(bounding_box[1], tmp_bounding_box[1], 0)
+                bounding_box[2] = update_function(bounding_box[2], tmp_bounding_box[2], 1)
+                bounding_box[3] = update_function(bounding_box[3], tmp_bounding_box[3], 1)
+            bounding_box_set.append(bounding_box)
+
+            # update start point
+            start_point = len(data['features'])
+    
+    return output_data, bounding_box_set, name_num_list, roadFile
+# =======================================
+def main():
+    # declare variables and arguments
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--folderPath', type = str, default='', help='path to an input folder')
+
+    args = parser.parse_args()
+    file_path = args.folderPath
+
+    # find an initial bounding box given all geometries
+    final_BB = None
+    entire_data, out_BB, out_name_num, road_file = bounding_box_process(file_path)
+
+    if len(out_BB) == 1:
```

Comment:
When is this not true?

Reply:
when there are multiple Geojson files in a directory, the condition is not true.